### ISSUE TITLE
The Squiz.Commenting.FunctionComment sniff doesn't detect incorrect types when @return tag has description

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#1018]
Example code:

``` php
/**
 * Return description function + mixed return.
 *
 * @return string|int This is a description.
 */
public function myFunction() {
    return 5;
}
```

In above example the replacement of `string|int` into `string|integer` should be suggested, but this doesn't happen because instead of `int` the `int This is a description.` is parsed as data type.

Fixing this can be problematic, because on of supported data type format includes spaces: `array(int => string)` for example.

Most likely `PHP_CodeSniffer::suggestType` method needs to be changed to allow understanding that case OR a code added to separate description from return type as done for `@param` tag.

Related issue: #1015

**Comment by gsherwood** [comment]
Thanks for the report. Fixed now.

**Comment by aik099** [comment]
Unfortunately you've fixed it in a way that breaks `array(int => string)` types since it does explode by space. This way `@return int|array(int => string) description` and `@return int|array(int => string)` won't work.

**Comment by aik099** [comment]
Actually if you had auto-fixed file in place you've seen that `array(...)` types aren't properly auto-fixed after this change.

**Comment by gsherwood** [comment]
Yep, broken for that code. Thanks for letting me know.

**Comment by gsherwood** [comment]
I way too busy to fix this array issue right now, especially trying to put in an auto-fix for it, so I've just reverted the change. I can't see this getting fixed any time soon.

**Comment by aik099** [comment]
That is correct decision. Thank you.

I personally had no idea, that `array(type => type)` format is even possible before looking at `suggestType` method. That's why I just reported the issue instead of fixing it.

**Comment by jrfnl** [comment]
This issue should be fixed by PR #1310
